src/Modules/AdminDashboard/Componenets/Rooms/RoomsData/RoomsData.tsx

```diff
@@ -1,3 +1,462 @@
+import { CloudUpload } from "@mui/icons-material";
+import {
+  Box,
+  Button,
+  CircularProgress,
+  FormControl,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+  Typography,
+} from "@mui/material";
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { useLocation, useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import { useAuth } from "../../../../../Context/AuthContext/AuthContext";
+import {
+  CreateRoom,
+  FacilitiesInterface,
+} from "../../../../../Interfaces/interFaces";
+import { getBaseUrl } from "../../../../../Utils/Utils";
+import style from "../Rooms.module.css";
+
 export default function RoomsData() {
-  return <div>RoomsData</div>;
+  const [selectedImages, setSelectedImages] = useState<string[]>([]);
+  const [spinner, setSpinner] = useState<boolean>(false);
+  const [isClicked, setIsClicked] = useState<boolean>(false);
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [facilities, setFacilities] = useState<FacilitiesInterface[]>([]);
+  const { requestHeaders } = useAuth();
+  const {
+    register,
+    handleSubmit,
+    control,
+    reset,
+    formState: { errors },
+  } = useForm<CreateRoom>();
+  const roomData = location.state?.roomData;
+  const resetFormValue = () => {
+    if (roomData) {
+      reset({
+        roomNumber: roomData.roomNumber,
+        price: roomData.price,
+        capacity: roomData.capacity,
+        discount: roomData.discount,
+        facilities: roomData.facilities.map(
+          (fac: FacilitiesInterface) => fac._id
+        ),
+      });
+    }
+  };
+
+  const getFacilities = async () => {
+    try {
+      const { data } = await axios.get(
+        `${getBaseUrl()}/api/v0/admin/room-facilities?page=1&size=1000`,
+        {
+          headers: requestHeaders,
+        }
+      );
+      setFacilities(data.data.facilities);
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response) {
+        toast.error(error.response.data.message || "fail get facilities");
+      }
+    }
+  };
+  const onSubmit = async (data: CreateRoom) => {
+    setSpinner(true);
+    setIsClicked(true);
```

Comment:
isClicked is less meaningfull, what has been clicked? We can not deduce that unless we read the code. I just suggest renaming this variable to a one that explains its real purpose